### PR TITLE
Fixed where annotation with like and a parameter with wildcard

### PR DESCRIPTION
**Pull request by octante**
This PR replaces las PR for like annotations and includes unit testing of RequestParameterProvider. Rebased with master.

### CODE REVIEW

**Comment by mmoreram** [comment]
@octante Can you rebase please? I've merged your last PR ^^

**Comment by octante** [comment]
Rebase is done, I hope all is ok :)

Thanks!!

File: Resolver/Paginator/PaginatorWheresEvaluator.php [hunk 1]
```diff
@@ -83,4 +87,37 @@ public function evaluate(
...
+     * Add wildcards to query if necessary
+     *
+     * @param string    $annotationWhereParameter   Where from annotation
+     * @param string    $whereValue                 Where replaced with request parameters
+     *
+     * @return string
+     */
+    private function addWildcards($annotationWhereParameter, $whereValue)
+    {
+        $hasInitialPercentage = (strpos($annotationWhereParameter, '%') === 0) ? true : false;
```
Comment (mmoreram) [inline Resolver/Paginator/PaginatorWheresEvaluator.php]: You can transform this

``` php
(strpos($annotationWhereParameter, '%') === 0) ? true : false;
```

to this

``` php
strpos($annotationWhereParameter, '%') === 0);
```

File: Resolver/Paginator/PaginatorWheresEvaluator.php [hunk 2]
```diff
@@ -83,4 +87,37 @@ public function evaluate(
...
+        $hasEndPercentage = (substr($annotationWhereParameter, -1) == '%') ? true : false;
```
Comment (mmoreram) [inline Resolver/Paginator/PaginatorWheresEvaluator.php]: Same here

File: Resolver/Paginator/PaginatorWheresEvaluator.php (hunk 1 above)
Comment (octante) [inline Resolver/Paginator/PaginatorWheresEvaluator.php]: Done! :D

2015-05-25 14:28 GMT+02:00 Marc Morera <notifications@github.com>:

> In Resolver/Paginator/PaginatorWheresEvaluator.php
> <https://github.com/mmoreram/ControllerExtraBundle/pull/68#discussion_r30977363>
> :
>
> > +    private function clearWildcards($where)
> > +    {
> > +        return trim($where, '%');
> > +    }
> > +
> > +    /**
> > +     * Add wildcards to query if necessary
> > +     *
> > +     * @param string    $annotationWhereParameter   Where from annotation
> > +     * @param string    $whereValue                 Where replaced with request parameters
> > +     *
> > +     * @return string
> > +     */
> > +    private function addWildcards($annotationWhereParameter, $whereValue)
> > +    {
> > +        $hasInitialPercentage = (strpos($annotationWhereParameter, '%') === 0) ? true : false;
>
> You can transform this
>
> (strpos($annotationWhereParameter, '%') === 0) ? true : false;
>
> to this
>
> strpos($annotationWhereParameter, '%') === 0);
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mmoreram/ControllerExtraBundle/pull/68/files#r30977363>
> .
>



-- 
Issel Guberna
Twitter: @octante
http://www.e-octante.com

File: Tests/UnitTest/Provider/RequestParameterProvider.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+ * file that was distributed with this source code.
+ *
+ * Feel free to edit as you please, and have fun.
+ *
+ * @author Marc Morera <yuhu@mmoreram.com>
+ */
+namespace Mmoreram\ControllerExtraBundle\Tests\UnitTest\Provider;
+use PHPUnit_Framework_TestCase;
+use Mmoreram\ControllerExtraBundle\Provider\RequestParameterProvider;
+class RequestParameterProviderTest extends PHPUnit_Framework_TestCase
```
Comment (mmoreram) [inline Tests/UnitTest/Provider/RequestParameterProvider.php]: Add PHPDoc to this class, and add some spaces here

File: Tests/UnitTest/Provider/RequestParameterProvider.php [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    {
+        $parameterBagMock   = $this->getParameterBagMock();
+        $requestMock        = $this->getRequestMock($parameterBagMock);
+        $requestStackMock   = $this->getRequestStackMock($requestMock);
+        $sut = new RequestParameterProvider($requestStackMock);
+        $sut->setRequestType(RequestParameterProvider::CURRENT_REQUEST);
+        $parameterValue = $sut->getParameterValue($parameter);
+        $this->assertEquals($result, $parameterValue);
+    }
+    /**
```
Comment (mmoreram) [inline Tests/UnitTest/Provider/RequestParameterProvider.php]: Spaces between methods

**Comment by mmoreram** [comment]
@octante Could you please apply the formatters? Thanks :D:D

**Comment by octante** [comment]
I have applied spaces, I don't know why this file is without format... strange..

Thanks!!

**Comment by octante** [comment]
I haven't applied bin/php-cs-fixer fix because is not in composer file, maybe can we add it?

**Comment by mmoreram** [comment]
@octante Have you applied the php-cs-fixer?

`php bin/php-cs-fixer fix`

**Comment by octante** [comment]
I'm adding php-cs-fixer in composer :)

**Comment by octante** [comment]
I'm applying cs-fixer and I have a lot of changes, for example, arrays to [], and so on. Sure I want to upload it?

**Comment by octante** [comment]
Upated :)

**Comment by octante** [comment]
I have removed short array syntax in php_cs file and applied again. Now must be compatible with php 5.3

**Pull request merged**